app/src/main/java/com/naman14/timberx/ui/adapters/FolderAdapter.kt

```diff
@@ -85,46 +80,46 @@ class FolderAdapter(private val mContext: Activity) : RecyclerView.Adapter<Folde
     }
 
     override fun getItemCount(): Int {
-        return mFileSet?.size ?: 0
-    }
-
-    fun goUpAsync(): Boolean {
-        if (mRoot == null || mBusy) {
-            return false
-        }
-        val parent = mRoot!!.parentFile
-        return if (parent != null && parent.canRead()) {
-            updateDataSetAsync(parent)
-        } else {
-            false
-        }
+        return files?.size ?: 0
     }
 
     fun updateDataSetAsync(newRoot: File): Boolean {
-        if (mBusy) {
+        if (isBusy) {
             return false
         }
         if (".." == newRoot.name) {
             goUpAsync()
             return false
         }
-        mRoot = newRoot
-        NavigateTask().executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR, mRoot)
+        rootFolder = newRoot
+        NavigateTask().executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR, rootFolder)
         return true
     }
 
+    private fun goUpAsync(): Boolean {
+        if (rootFolder == null || isBusy) {
+            return false
+        }
+        val parent = rootFolder!!.parentFile
```

Comment:
you can easily kill the `!!` if you do a `val rootFolder = rootFolder` above the first `if` 3 lines above

Reply:
@Zhuinden also probably out of scope for this PR

Reply:
Not really, it's literally +1 line

Reply:
Maybe not since this is a cleanup PR